var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -30,16 +30,32 @@ class Mesa(AutotoolsPackage):
     specification - a system for rendering interactive 3D graphics."""
 
     homepage = "http://www.mesa3d.org"
-    url      = "http://ftp.iij.ad.jp/pub/X11/x.org/pub/mesa/12.0.3/mesa-12.0.3.tar.gz"
+    base_url = "ftp://ftp.freedesktop.org/pub/mesa/"
+    url      = base_url + "12.0.3/mesa-12.0.3.tar.gz"
 
+    version('13.0.2', 'cb828f9f68e0cd6c7ef30b9c67aa7cf2')
+    version('13.0.1', '8415d4bb7837e6cfb0c819fdd19a643b')
+    version('13.0.0', '7205edb90d0396dc26d049fa495f6fd1')
     version('12.0.3', '60c5f9897ddc38b46f8144c7366e84ad')
 
+    variant('gallium', default=False, description="compile with gallium llvm sw rendering")
+    variant('drm', default=True, description="compile with drm")
```

Comment:
> I' ll try to use it to reduce dependencies, like libdrm that now is a unconditional dep. I will test on server without in and then plan to leave it under drm variant.

i don't know what `drm` is, but if it is indeed unconditional dependency, you should keep it as such regardless of whether or not it's provided by the system. If someone wants to build with system-provided `libdrm`, then they should configure external package in `packages.yaml`.